core/lib/workarea/ext/jbuilder/jbuilder_cache.rb

```diff
@@ -0,0 +1,31 @@
+require 'jbuilder/jbuilder_template'
+
+decorate JbuilderTemplate do
+  def _cache_fragment_for(key, options)
+    return yield if workarea_admin?
+
+    super
+  end
+
+  def _cache_key(key, options)
+    super.tap do |result|
+      result << workarea_cache_varies if workarea_cache_varies.present?
+    end
+  end
+
+  private
+
+  def workarea_admin?
+    @context&.controller&.current_user&.admin?
+  rescue ::RuntimeError
+    false
+  end
+
+  def workarea_cache_varies
+    workarea_request_env['workarea.cache_varies']
+  end
+
+  def workarea_request_env
+    @context.controller.request.env || {}
```

Comment:
This `env` isn't necessarily set in places like helper tests, or when rendering a JSON response from a controller programmatically (using `ActionController::Base.render`), so we gotta be extra safe here.